mysql/resultset_helper.go

```diff
@@ -39,13 +43,53 @@ func FormatTextValue(value interface{}) ([]byte, error) {
 		return v, nil
 	case string:
 		return utils.StringToByteSlice(v), nil
+	case time.Time:
+		return utils.StringToByteSlice(v.Format(time.DateTime)), nil
 	case nil:
 		return nil, nil
 	default:
 		return nil, errors.Errorf("invalid type %T", value)
 	}
 }
 
+func toBinaryDateTime(t time.Time) ([]byte, error) {
+	var buf bytes.Buffer
+
+	if t.IsZero() {
+		return nil, fmt.Errorf("zero time")
```

Comment:
I think MySQL can support insert zero time. So this library should also allow it?

```mysql
mysql> create table t (t time);
Query OK, 0 rows affected (0.07 sec)

mysql> insert into t values('0000-00-00 00:00:00');
Query OK, 1 row affected (0.01 sec)

mysql> select * from t;
+----------+
| t        |
+----------+
| 00:00:00 |
+----------+
1 row in set (0.00 sec)
```